File: Python/Problem263A_BeautifulMatrix.py
def main():
    for row in range(0,6):
        col1, col2, col3, col4, col5 = map(int, input().split())
        if col1 == 1:
            print(2 + abs(2-row))
            break
        elif col2 == 1:
            print(1 + abs(2-row))
            break
        elif col3 == 1:
            print(abs(2-row))
            break
        elif col4 == 1:
            print(1 + abs(2-row))
            break
        elif col5 == 1:
            print(2 + abs(2-row))
            break
        else:
            continue

if __name__ == "__main__":
    main()

File: C++/Problem263A_BeautifulMatrix.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main(){
    int col_1,col_2,col_3,col_4,col_5;
    
    for(int row = 1; row < 6; row++){
        cin >> col_1 >> col_2 >> col_3 >> col_4 >> col_5;
        if(col_1 == 1){ 
            cout << 2 + abs(row-3);
            break;
        }
        else if(col_2 == 1){
            cout << 1 + abs(row-3);
            break;
        }
        else if(col_3 == 1){
            cout << abs(row-3);
            break;
        }
        else if(col_4 == 1){
            cout << 1 + abs(row-3);
            break;
        }
        else if(col_5 == 1){
            cout << 2 + abs(row-3);
            break;
        }
        else{
            continue;
        }
    }
    return 0;
}